semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -542,21 +588,35 @@ let check_tainted_instr env instr : Taints.t * Lval_env.t =
           args
           |> List.fold_left_map
                (fun lval_env arg ->
-                 check_expr { env with lval_env } arg |> Common2.swap)
+                 let taints, lval_env =
+                   check_expr { env with lval_env }
+                     (IL_lvalue_helpers.exp_of_arg arg)
+                 in
+                 match arg with
+                 | Unnamed _ -> (lval_env, Left taints)
+                 | Named (id, _) -> (lval_env, Right (id, taints)))
                env.lval_env
           |> Common2.swap
         in
+        let all_taints =
+          args_taints
+          |> Common.map (function
+               | Left taints -> taints
```

Comment:
I would use variant types here instead of `Left` and `Right`, makes the code easier to read. The only inconvenience is that you can't use `partition_either`, but using a regular `partition` isn't very verbose anyways.